### PR TITLE
feat: add new lines inside TOC and link to bitdowntoc sources

### PR DESCRIPTION
**Pull request by derlin**
This commit:
- adds a new line after the TOC start comment and before the TOC end comment.
- changes the TOC start to add a link to bitdowntoc.
- adds a new line after the TOC if there is no toc marker/existing toc

The result is thus:

```
{%- # TOC start (generated with https://github.com/derlin/bitdowntoc) -%}

- [Introduction](#introduction)
  - [S3 specific outputs](#s3-specific-outputs)
- [Conclusion](#conclusion)

{%- # TOC end -%}

```

Resolves: #9
Resolves: #10

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,7 +24,8 @@ It supports English, French, and most **Latin languages**, but not Cyrillic or C
 
 TOC (generated by this tool, duh, using the `github` profile):
 
-<!-- TOC start -->
+<!-- TOC start (generated with https://github.com/derlin/bitdowntoc) -->
```
Comment (bcouetil) [inline README.md:27]: I would prefer the *.io URL, for us to click on the link and regenerate directly.

But I confess that github.com click and, there, github.io click, is not that difficult.

File: src/commonTest/kotlin/ch.derlin.bitdowntoc/GenerateTest.kt [hunk 2]
```diff
@@ -13,9 +13,12 @@ class GenerateTest {
 
         assertEquals(
             """
-            <!-- TOC start -->
+            <!-- TOC start (generated with $BITDOWNTOC_URL) -->
+            
             - [heading](#heading)
+            
             <!-- TOC end -->
+
```
Comment (bcouetil) [inline src/commonTest/kotlin/ch.derlin.bitdowntoc/GenerateTest.kt:21]: Perfect

**Review by bcouetil — COMMENTED** [review]
One minor change if you want, but not mandatory.

Many thanks for the improvement 🙏

File: README.md (hunk 1 above)
Comment (derlin) [reply README.md:27]: I hesitated as well,  but there is also a CLI (jar) available. Moreover, I am thinking of maybe moving the github.io website to one of my domains (if you have an opinion, I am all ears). The GitHub URL is thus less likely to change. The implementation makes it possible to change our minds later though :)

**Comment by derlin** [comment]
Thank *you* for using bitdowntoc and suggesting improvements!

File: README.md (hunk 1 above)
Comment (bcouetil) [reply README.md:27]: OK that makes sense, let's leave it that way ✌️

**Pull request merged**